src/sql/src/plan/statement/ddl.rs

```diff
@@ -404,38 +403,18 @@ pub fn plan_create_source(
                 _ => unsupported!(format!("AWS Resource type: {:#?}", arn.resource)),
             };
 
-            let region: Region = match arn.region {
-                Some(region) => match region.parse() {
-                    Ok(region) => {
-                        // ignore the endpoint option if we're pointing at a
-                        // valid, non-custom AWS region
-                        with_options.remove("endpoint");
-                        region
-                    }
-                    Err(e) => {
-                        // Region's fromstr doesn't support parsing custom regions.
-                        // If a Kinesis stream's ARN indicates it exists in a custom
-                        // region, support it iff a valid endpoint for the stream
-                        // is also provided.
-                        match with_options.remove("endpoint") {
-                            Some(Value::String(endpoint)) => Region::Custom {
-                                name: region,
-                                endpoint,
-                            },
-                            _ => bail!(
-                                "Unable to parse AWS region: {}. If providing a custom \
-                                        region, an `endpoint` option must also be provided",
-                                e
-                            ),
-                        }
-                    }
-                },
-                None => bail!("Provided ARN does not include an AWS region"),
-            };
+            let region = arn
+                .region
+                .ok_or_else(|| anyhow!("Provided ARN does not include an AWS region"))?;
 
+            let aws_info = aws_connect_info(&mut with_options, Some(region))?;
+            futures::executor::block_on(aws_util::aws::validate_credentials(
+                aws_info.clone(),
+                Duration::from_secs(1),
+            ))?;
```

Comment:
Eek, blocking in the executor like this is not ok! It'll prevent any other SQL query from getting processed for 1s.